libraries/AP_Camera/AP_Camera_Backend.cpp

```diff
@@ -415,4 +449,178 @@ void AP_Camera_Backend::log_picture()
+    if (!_copy_json_field_double(obj, "focal_length", camera_info.msg.focal_length)) goto err;
+    if (!_copy_json_field_double(obj, "sensor_size_h", camera_info.msg.sensor_size_h)) goto err;
+    if (!_copy_json_field_double(obj, "sensor_size_v", camera_info.msg.sensor_size_v)) goto err;
+    if (!_copy_json_field_double(obj, "resolution_h", camera_info.msg.resolution_h)) goto err;
+    if (!_copy_json_field_double(obj, "resolution_v", camera_info.msg.resolution_v)) goto err;
+    if (!_copy_json_field_double(obj, "flags", camera_info.msg.flags)) goto err;
+    if (!_copy_json_field_double(obj, "cam_definition_version", camera_info.msg.cam_definition_version)) goto err;
+
+    if (!_copy_json_field_string(obj, "vendor_name", (char*)camera_info.msg.vendor_name, sizeof(camera_info.msg.vendor_name))) goto err;
+    if (!_copy_json_field_string(obj, "model_name", (char*)camera_info.msg.model_name, sizeof(camera_info.msg.model_name))) goto err;
+    if (!_copy_json_field_string(obj, "cam_definition_uri", camera_info.msg.cam_definition_uri, sizeof(camera_info.msg.cam_definition_uri))) goto err;
+
+    // Populate the fields that shouldn't change.
+    camera_info.msg.time_boot_ms = AP_HAL::millis();
+    // To disambiguate when we have multiple cameras, the lens_id is populated
+    // with the instance number.
+    camera_info.msg.lens_id = _instance + 1;
+    camera_info.msg.gimbal_device_id = get_gimbal_device_id();
+
+    // If we got this far, then we've got all the required fields.
+    camera_info.is_valid = true;
+
+err:
+    delete obj;
+
+    if (camera_info.is_valid) {
+        debug("Loaded camera_info from '%s'", filename);
+        debug("    camera_info.msg.time_boot_ms=%u", camera_info.msg.time_boot_ms);
+        debug("    camera_info.msg.vendor_name='%s'", camera_info.msg.vendor_name);
+        debug("    camera_info.msg.model_name='%s'", camera_info.msg.model_name);
+        debug("    camera_info.msg.firmware_version=%u", camera_info.msg.firmware_version);
+        debug("    camera_info.msg.focal_length=%f", camera_info.msg.focal_length);
+        debug("    camera_info.msg.sensor_size_h=%f", camera_info.msg.sensor_size_h);
+        debug("    camera_info.msg.sensor_size_v=%f", camera_info.msg.sensor_size_v);
+        debug("    camera_info.msg.resolution_h=%u", camera_info.msg.resolution_h);
+        debug("    camera_info.msg.resolution_v=%u", camera_info.msg.resolution_v);
+        debug("    camera_info.msg.lens_id=%u", camera_info.msg.lens_id);
+        debug("    camera_info.msg.flags=%u", camera_info.msg.flags);
+        debug("    camera_info.msg.cam_definition_version=%u", camera_info.msg.cam_definition_version);
+        debug("    camera_info.msg.cam_definition_uri='%s'", camera_info.msg.cam_definition_uri);
+        debug("    camera_info.msg.gimbal_device_id=%u", camera_info.msg.gimbal_device_id);
+    }
+}
+
+#undef GET_JSON_DOUBLE
+#undef GET_JSON_STR
+
```

Comment:
Were these replaced by the template function?

Reply:
Yes, they were. Fixed.